packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.base.tsx

```diff
@@ -48,7 +48,7 @@ export class ChoiceGroupBase extends React.Component<IChoiceGroupProps, IChoiceG
       keyFocused: undefined
     };
 
-    this._id = getId('ChoiceGroup');
+    this._id = props.id || getId('ChoiceGroup');
```

Comment:
This would make `this._id` not unique anymore. 

I think a better approach will be to not modify `this._id`, but use `getNativeProps` from Utilities and insert the generated nativeProps to the root element using a spread operator.

Reply:
This approach I found in other fabric components. Seems like only ChoiceGroup was unique in a way how ID was resolved. For example:

- ComboBox.tsx line 187 this._id = props.id || getId('ComboBox');
- DatePicker.base.tsx line 85 this._id = props.id || getId('DatePicker');
- DropDown.base.tsx line 97 this._id = props.id || getId('Dropdown');

the way how id is assigned to a particular dom element is very random per component. In most of the places I saw it if suffixed with "-label", "-input", "-option" etc. Some elements have use id as it is (Dropdown.base.tsx:255, Checkbox.base.tsx:104)

Reply:
reverted this logic in the latest commit

Reply:
Oh I see, we should establish a consistent pattern.